ovs/extensions/healthcheck/expose_to_cli.py

```diff
@@ -0,0 +1,324 @@
+        self.function = func
+        func.module_name = self.module_name
+        func.method_name = self.method_name
+        return func
+    
+    
+class CLIRunner(object):
+    """
+    Runs a method exposed by the expose_to_cli decorator. Serves as a base for all extensions using expose_to_cli
+    """
+    logger = LogHandler.get("ovs", "clirunner")
+    OPTIONAL_ARGUMENTS = ["--help"]
+    START_PATH = ''.join([os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir))])
+
+    def __init__(self):
+        pass
+
+    @classmethod
+    def run_method(cls, *args):
+        """
+        Executes the given method
+        :return:
+        """
+        module_name = ''
+        method_name = ''
+        # Extract option arguments
+        optional_arguments = [arg for arg in cls.OPTIONAL_ARGUMENTS if arg in args]
+        args = [arg for arg in args if arg not in optional_arguments]
+        # Check for remaining arguments
+        if len(args) >= 1:
+            module_name = args[0]
+        if len(args) >= 2:
+            method_name = args[1]
+        # Determine method to execute
+        if (method_name and not module_name) or "--help" in optional_arguments:
+            print "Both the module name and the method name must be specified.".format(method_name, module_name)
+            return cls.print_help(module_name, method_name)
+        # Find the required method and execute it
+        discovered_data = cls._discover_methods(module_name, method_name)
+        if discovered_data is None:
+            print "Found no method {0} for module {1}".format(method_name, module_name)
+            return cls.print_help(module_name)
+        interrupted = False
+        try:
+            for exposed_module, exposed_methods in discovered_data.iteritems():
+                if module_name and exposed_module != module_name:
+                    continue
+                for method_data in exposed_methods:
+                    if not method_name or method_data['method_name'] == method_name:
+                        mod = imp.load_source(method_data['module_name'], method_data['location'])
+                        cl = getattr(mod, method_data['class'])()
+                        if '--help' in optional_arguments:
+                            print getattr(cl, method_data['function']).__doc__
+                            return
+                        # Execute method
+                        try:
+                            getattr(cl, method_data['function'])(*args)
+                        except Exception:
+                            cls.logger.exception('Error during execution of {0}.{1}'.format(cl, method_data['function']))
+        except KeyboardInterrupt:
+            interrupted = True
+            cls.logger.warning('Caught keyboard interrupt. Output may be incomplete!')
+        if interrupted is False:
+            print "Found no methods for module {0}".format(module_name)
+            return cls.print_help()
```

Comment:
I'm not sure about this either. Wont this be printed every time?  Unless you do a KeyboardInterrupt